src/test/java/picard/fingerprint/CheckFingerprintTest.java

```diff
@@ -0,0 +1,108 @@
+package picard.fingerprint;
+
+import htsjdk.samtools.metrics.MetricsFile;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.tribble.TribbleException;
+import org.testng.Assert;
+import org.testng.annotations.*;
+import picard.cmdline.CommandLineProgramTest;
+
+import java.io.File;
+import java.io.IOException;
+
+public class CheckFingerprintTest extends CommandLineProgramTest {
+
+    private static final File TEST_DATA_DIR = new File("testdata/picard/fingerprint/");
+    private static final File SUBSETTED_HAPLOTYPE_DATABASE_FOR_TESTING =
+            new File(TEST_DATA_DIR, "Homo_sapiens_assembly19.haplotype_database.subset.txt");
+    private static final File TEST_INPUT_VCF1 = new File(TEST_DATA_DIR, "NA12892.vcf");
+    private static final File TEST_INPUT_VCF_EMPTY = new File(TEST_DATA_DIR, "/tempCheckFPDir/void.vcf");
+    private static final File TEST_INPUT_VCF_NO_FILE = new File(TEST_DATA_DIR, "noFile.vcf");
+    private static final File TEST_OUTPUT = new File(TEST_DATA_DIR + "/tempCheckFPDir/otp.fp");
+    private static final File TEST_GENOTYPES_VCF1 = new File(TEST_DATA_DIR, "NA12892.g.vcf");
+    private static final File TEST_GENOTYPES_VCF_NO_FILE = new File(TEST_DATA_DIR, "noFile.g.vcf");
+    private static final File RESULT_EXAMPLE_SUMMARY =
+            new File(TEST_DATA_DIR, "fingerprinting_summary_metrics.example");
+    private static final File RESULT_EXAMPLE_DETAIL =
+            new File(TEST_DATA_DIR, "fingerprinting_detail_metrics.example");
+    private final File tempFolder = new File(TEST_DATA_DIR + "/tempCheckFPDir/");
+
+    @BeforeClass
+    private void mkTemp() {
+        if (!tempFolder.exists()) {
+            tempFolder.mkdir();
+        }
+    }
+
+    @AfterClass
+    private void rmTemp() {
+        IOUtil.deleteDirectoryTree(tempFolder);
+    }
+
+    @DataProvider(name = "badData")
+    public Object[][] badData() {
+        return new Object[][]{
+                {TEST_INPUT_VCF_EMPTY, TEST_OUTPUT,
+                        TEST_GENOTYPES_VCF1, SUBSETTED_HAPLOTYPE_DATABASE_FOR_TESTING},
+                {TEST_INPUT_VCF_NO_FILE, TEST_OUTPUT,
+                        TEST_GENOTYPES_VCF1, SUBSETTED_HAPLOTYPE_DATABASE_FOR_TESTING},
+                {TEST_INPUT_VCF1, TEST_OUTPUT,
+                        TEST_GENOTYPES_VCF_NO_FILE, SUBSETTED_HAPLOTYPE_DATABASE_FOR_TESTING},
+                {TEST_INPUT_VCF_NO_FILE, TEST_OUTPUT,
+                        TEST_GENOTYPES_VCF_NO_FILE, SUBSETTED_HAPLOTYPE_DATABASE_FOR_TESTING},
+        };
+    }
+    @Test
+    public void testBaseOutput() throws IOException {
+        String[] args = new String[] {
+                "I=" + TEST_INPUT_VCF1,
+                "O=" + TEST_OUTPUT,
+                "G=" + TEST_GENOTYPES_VCF1,
+                "H=" + SUBSETTED_HAPLOTYPE_DATABASE_FOR_TESTING
+        };
+        Assert.assertEquals(runPicardCommandLine(args), 0);
+        Assert.assertEquals(checkResult(
```

Comment:
I think it would be better to have two asserts, one for each metrics file....that way when something breaks it's easier to understand where the probem is

Reply:
Done!